### ISSUE TITLE
Install on Ubuntu 18.04 

### ISSUE THREAD

**Issue by gjduran** [ltb-project/self-service-password#293]
Whatever I've tried I can't get SSP to work on Ubuntu 18.04. It refuses to connect to LDAP no matter what settings I use. Is there a non-documented dependency? Has anyone had any success on Ubuntu 18.04 and would like to share what it took to get it done?

**Comment by coudot** [comment]
Did you install PHP LDAP package?

**Comment by gjduran** [comment]
Yes I did. I'm using LDAP with other apps (Zarafa/Kopano, Odoo, SVN etc) and that working fine.
I encountered that some dependencies are no longer supported in Ubuntu 18.04 but I followed the work around's available in the various threats. The only one which is missing is authz_ldap but that seem to be replaced with authnz_ldap and I'm not sure whether that is causing the problem.
Some output of the ssp_error.log:
[ssp_error.log]
[Fri Apr 26 10:30:18.340545 2019] [php7:warn] [pid 13194] [client 192.168.80.21:52358] PHP Warning:  ldap_bind(): Unable to bind to server: Invalid credentials in /usr/share/self-service-password/pages/change.php on line 80, referer: https://ssp.xxx-xxxxx.com/
[Fri Apr 26 10:30:18.340595 2019] [php7:notice] [pid 13194] [client 192.168.80.21:52358] LDAP - Bind error 49  (Invalid credentials), referer: https://ssp.xxx-xxxxx.com/
[Fri Apr 26 10:30:18.340607 2019] [php7:notice] [pid 13194] [client 192.168.80.21:52358] PHP Notice:  Undefined variable: obscure_failure_messages in /usr/share/self-service-password/pages/change.php on line 186, referer: https://ssp.xxx-xxxxx.com/
[Fri Apr 26 10:30:18.340619 2019] [php7:warn] [pid 13194] [client 192.168.80.21:52358] PHP Warning:  in_array() expects parameter 2 to be array, null given in /usr/share/self-service-password/pages/change.php on line 186, referer: https://ssp.xxx-xxxxx.com/
[Fri Apr 26 10:30:18.340795 2019] [php7:notice] [pid 13194] [client 192.168.80.21:52358] PHP Notice:  Undefined variable: display_posthook_error in /usr/share/self-service-password/pages/change.php on line 193, referer: https://ssp.xxx-xxxxx.com/

**Comment by coudot** [comment]
The error seems clear: `Unable to bind to server: Invalid credentials`

Check the Bind DN and Bind Password that is configured.

**Comment by gjduran** [comment]
This is my config...
[config.inc.local.php]
$ldap_url = "ldap://10.67.1.2:389";
$ldap_starttls = false;
$ldap_binddn = "dc=xxx-xxxxx,dc=com";
$ldap_bindpw = "secret"
$ldap_base = "ou=people,dc=xxx-xxxxx,dc=com";
$ldap_login_attribute = "uid";
$ldap_fullname_attribute = "cn";
$ldap_filter = "(uid={login})";

**Comment by gjduran** [comment]
Thanks for your prompt response yesterday which motivated me to dive into it once more and with result, found the problem!!! 
If the php statement "<?php" at the beginning of the secondary config file is missing, the new variables are ignored and leaves you with the default settings. Also in this case, the content of this config.inc.local.php file, is displayed on the webpage but hardly visible due to the default background. I figured this out by ignoring the advise to use config.inc.local.php for custom settings and edited the original config.inc.php instead. Than I saw that the app box was drawn more towards the top of the screen and than I discovered that the content of config.inc.local.php was drawn first but hardly visible due to the dark background. So I disabled the background image in config.inc.php to be sure and start to toggle "<?php" in config.inc.local.php on and off and voila, problem solved.
Reading through all the threads it looks that many more are facing the same problem....
